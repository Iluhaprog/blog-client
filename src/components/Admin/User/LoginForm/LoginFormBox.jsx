import styled from 'styled-components';
import {connect} from 'react-redux';
import {themes} from '../../../../store/settings/settingsReducer';

let LoginFormBox = styled.div`
  position: absolute;
  top: 0; bottom: 0; left: 0; right: 0;
  max-width: 300px;
  height: 300px;
  margin: auto;
  background-color: ${({theme}) => {
    return theme === themes.DARK ? '#212529' : '#f8f9fa';
  }};
  box-shadow: 0 10px 15px rgba(0, 0, 0, 0.3);
  border-radius: 10px;
  padding: 15px 0;
`;

const mapStateToProps = (state) => ({
  theme: state.settings.theme,
});

const mapDispatchToProps = () => ({});

LoginFormBox = connect(
    mapStateToProps,
    mapDispatchToProps,
)(LoginFormBox);

export {LoginFormBox};
