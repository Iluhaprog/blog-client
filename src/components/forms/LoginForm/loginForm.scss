@import '../../../vars.scss';

.login-form {
    h1 {
        font-size: 24px;
        font-weight: 300;
        color: $nav-bg;
        margin-bottom: 65px;
    }

    .form__box {
        margin-bottom: 35px;
    }
}

.link {
    display: block;
    width: 100%;
    text-align: center;
    margin: 15px 0;
}

@keyframes anim-err{
    0% {
        letter-spacing: 0.5px;
    }
    30% {
        letter-spacing: 1.5px;
    }
    90% {
        letter-spacing: 0.3px;
    }
    100% {
        letter-spacing: 0.5px;
    }
}

.error {
    display: block;
    width: 100%;
    color: $error;
    opacity: 0;
    transition: .3s;
    text-align: center;
    margin: 10px 0;
    letter-spacing: 0.5px;
    animation: anim-err 1s 2s infinite ease-in-out;

    &_visible {
        opacity: 1;
    }
}