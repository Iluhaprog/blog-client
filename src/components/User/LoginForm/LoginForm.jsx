import React from 'react';
import {Formik, Form, Field} from 'formik';
import {Badge, Button, Container, Row} from 'react-bootstrap';
import * as PropTypes from 'prop-types';
import {LoginFormBox} from './LoginFormBox';
import {connect} from 'react-redux';
import {login} from '../../../store/auth/authActions';
import {InputField} from '../../Field/Input';

let LoginForm = ({lang, login}) => {
  const submit = (values, actions) => {
    const {username, password} = values;
    login(username, password);
  };

  return (
    <LoginFormBox>
      <Formik
        initialValues={{
          username: '',
          password: '',
        }}
        onSubmit={submit}
      >
        <Form>
          <Container>
            <Row className='justify-content-center'>
              <h4>
                <Badge
                  variant={'secondary'}
                >
                  {lang.text.SIGN_IN}
                </Badge>
              </h4>
            </Row>
            <Field
              name='username'
              type={'text'}
              label={`${lang.label.LOGIN}:`}
              component={InputField}
              placeholder={`${lang.label.LOGIN}...`}
            />
            <Field
              name='password'
              type={'password'}
              label={`${lang.label.PASSWORD}:`}
              component={InputField}
              placeholder={`${lang.label.PASSWORD}...`}
            />
            <Row className='justify-content-center'>
              <Button
                style={{width: 'calc(100% - 30px)'}}
                variant={'primary'}
                type={'submit'}
              >
                {lang.text.SIGN_IN}
              </Button>
            </Row>
          </Container>
        </Form>
      </Formik>
    </LoginFormBox>
  );
};

LoginForm.propTypes = {
  login: PropTypes.func,
  lang: PropTypes.object,
};

const mapStateToProps = (state) => ({
  lang: state.settings.lang,
});

const mapDispatchToProps = (dispatch) => ({
  login: (username, password) => {
    dispatch(login(username, password));
  },
});

LoginForm = connect(mapStateToProps, mapDispatchToProps)(LoginForm);

export {LoginForm};
