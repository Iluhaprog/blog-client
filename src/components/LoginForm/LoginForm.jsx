import { reduxForm } from 'redux-form';
import { NavLink } from 'react-router-dom';
import FormBox from '../FormBox';
import FormElement from '../FormElement';
import SendButton from '../SendButton';
import './loginForm.scss';

function LoginForm(props) {
    const { err, handleSubmit } = props;
    return (
        <FormBox>
            <form onSubmit={handleSubmit} className='login-form'>
                <div className='column column_jc-c column_ai-c'>
                    <h1>Login</h1>
                    <FormElement 
                        label='Email' 
                        type='email' 
                        name='username' 
                        component='input'
                    />
                    <FormElement 
                        label='Password' 
                        type='password' 
                        name='password' 
                        component='input'
                        />
                    <SendButton text='Login'/>
                </div>
            </form>
            <NavLink to='/registration' className='link'>
                Registration
            </NavLink>
            <strong className={`error ${err.status === 403 && 'error_visible'}`}>
                Wrong email or password
            </strong>
        </FormBox>
    );
}

export default reduxForm({
    form: 'login',
})(LoginForm);