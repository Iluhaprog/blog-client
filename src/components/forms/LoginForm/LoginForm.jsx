import React from 'react';
import { reduxForm } from 'redux-form';
import { NavLink } from 'react-router-dom';
import { FormBox } from '../../boxes';
import { LabeledInput } from '../../formElements';
import { SendButton } from '../../buttons';
import './loginForm.scss';

function LoginForm(props) {
    const { err, handleSubmit } = props;
    return (
        <FormBox>
            <form onSubmit={handleSubmit} className='login-form'>
                <div className='column column_jc-c column_ai-c'>
                    <h1>Login</h1>
                    <LabeledInput 
                        label='Email' 
                        type='email' 
                        name='username' 
                        component='input'
                    />
                    <LabeledInput 
                        label='Password' 
                        type='password' 
                        name='password' 
                        component='input'
                        />
                    <SendButton text='Login'/>
                </div>
            </form>
            <NavLink to='/registration' className='link'>
                Registration
            </NavLink>
            <strong className={`error ${err.status === 403 && 'error_visible'}`}>
                Wrong email or password
            </strong>
        </FormBox>
    );
}

export default reduxForm({
    form: 'login',
})(LoginForm);