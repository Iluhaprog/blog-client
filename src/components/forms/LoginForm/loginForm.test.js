import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react';
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

import LoginForm from './LoginForm';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

const authorizedState = {
    authorized: true,
    status: 0,
    errorData: {},
};

const unauthorizedState = {
    authorized: false,
    status: 403,
    errorData: {
        msg: 'Forbidden',
    },
};

describe('Test view part of login form', () => {
    test('Test login form view without failure login', () => {
        const { status, errorData } = authorizedState;
        const store = mockStore(authorizedState);
        const { asFragment } = render(
            <Provider store={store}>
                <BrowserRouter>
                    <LoginForm err={{ status, errorData }} onSubmit={e => e} />
                </BrowserRouter>
            </Provider>
        );
        expect(asFragment()).toMatchSnapshot();
    });

    test('Test login form view with failure login', () => {
        const { status, errorData } = unauthorizedState;
        const store = mockStore(unauthorizedState);
        const { asFragment } = render(
            <Provider store={store}>
                <BrowserRouter>
                    <LoginForm err={{ status, errorData }} onSubmit={e => e} />
                </BrowserRouter>
            </Provider>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});

test('Test submit event for login form', async () => {
    const { status, errorData } = unauthorizedState;
    const handleSubmit = jest.fn();
    const store = mockStore(unauthorizedState);
    const { container } = render(
        <Provider store={store}>
            <BrowserRouter>
                <LoginForm err={{ status, errorData }} onSubmit={handleSubmit} />
            </BrowserRouter>
        </Provider>
    );
    fireEvent.submit(container.getElementsByTagName('form')[0]);
    await waitFor(() => {
        expect(handleSubmit).toHaveBeenCalledTimes(1);
    });
});