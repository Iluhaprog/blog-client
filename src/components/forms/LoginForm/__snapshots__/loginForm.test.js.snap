// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test view part of login form Test login form view with failure login 1`] = `
<DocumentFragment>
  <div
    class="form-box"
  >
    <form
      class="login-form"
    >
      <div
        class="column column_jc-c column_ai-c"
      >
        <h1>
          Login
        </h1>
        <div
          class="form__box row-reverse"
        >
          <input
            name="username"
            type="email"
            value=""
          />
          <label
            for="username"
          >
            Email
          </label>
        </div>
        <div
          class="form__box row-reverse"
        >
          <input
            name="password"
            type="password"
            value=""
          />
          <label
            for="password"
          >
            Password
          </label>
        </div>
        <button
          class="button send-button"
        >
          Login
        </button>
      </div>
    </form>
    <a
      class="link"
      href="/registration"
    >
      Registration
    </a>
    <strong
      class="error error_visible"
    >
      Wrong email or password
    </strong>
  </div>
</DocumentFragment>
`;

exports[`Test view part of login form Test login form view without failure login 1`] = `
<DocumentFragment>
  <div
    class="form-box"
  >
    <form
      class="login-form"
    >
      <div
        class="column column_jc-c column_ai-c"
      >
        <h1>
          Login
        </h1>
        <div
          class="form__box row-reverse"
        >
          <input
            name="username"
            type="email"
            value=""
          />
          <label
            for="username"
          >
            Email
          </label>
        </div>
        <div
          class="form__box row-reverse"
        >
          <input
            name="password"
            type="password"
            value=""
          />
          <label
            for="password"
          >
            Password
          </label>
        </div>
        <button
          class="button send-button"
        >
          Login
        </button>
      </div>
    </form>
    <a
      class="link"
      href="/registration"
    >
      Registration
    </a>
    <strong
      class="error false"
    >
      Wrong email or password
    </strong>
  </div>
</DocumentFragment>
`;
